src/app/products/products-list/products-list.component.html

```diff
@@ -0,0 +1,32 @@
+<div class="flex-container">
+  <div class="flex-container-left border rounded">
+    <app-product-filter (searchValue)="filterData($event)"></app-product-filter>
+    <br />
+    <ul class="list-group">
+      <li
+        class="list-group-item list-group-item-action"
+        *ngFor="let product of products | async"
+        (click)="onClick(product)"
+        [ngClass]="addActiveClass(product)"
+      >
+        <div [ngClass]="addRedWarningClass(product)">
```

Comment:
duże NIE dla funkcji w widokach. podwójne duze NIE NIE dla  funkcji w ngClass.
mozna to zrobic tak: [ngClass]="{'active': product.id === selectedProduct.id, 'warning':product.id === 0 }" To rozwiazanie duzo bardziej optymalne bo angular nie wykonuje tych porownan za kazdym razem a tylko wtedy kiedy zmienia sie zmienne. w przypadku funkcji wykonywane sa one zawsze jak tylko change detector jest uruchamiany - nawet w innym komponencie ktory jest na aktualnym widoku

Reply:
ok